nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/computation/log/ComputationRunner.java

```diff
@@ -0,0 +1,369 @@
+    protected volatile boolean drain = false;
+    protected final CountDownLatch assignmentLatch = new CountDownLatch(1);
+
+    protected Computation computation;
+    protected long counter = 0;
+    protected long inRecords = 0;
+    protected long inCheckpointRecords = 0;
+    protected long outRecords = 0;
+    protected final WatermarkMonotonicInterval lowWatermark = new WatermarkMonotonicInterval();
+    protected long lastReadTime = System.currentTimeMillis();
+    protected long lastTimerExecution = 0;
+    protected String threadName;
+
+    @SuppressWarnings("unchecked")
+    public ComputationRunner(Supplier<Computation> supplier, ComputationMetadataMapping metadata,
+                             List<LogPartition> defaultAssignment, LogManager logManager) {
+        this.supplier = supplier;
+        this.metadata = metadata;
+        this.logManager = logManager;
+        this.context = new ComputationContextImpl(metadata);
+        if (metadata.inputStreams().isEmpty()) {
+            this.tailer = null;
+            assignmentLatch.countDown();
+        } else if (logManager.supportSubscribe()) {
+            this.tailer = logManager.subscribe(metadata.name(), metadata.inputStreams(), this);
+        } else {
+            this.tailer = logManager.createTailer(metadata.name(), defaultAssignment);
+            assignmentLatch.countDown();
+        }
+    }
+
+    public void stop() {
+        log.debug(metadata.name() + ": Receives Stop signal");
+        stop = true;
+    }
+
+    public void drain() {
+        log.debug(metadata.name() + ": Receives Drain signal");
+        drain = true;
+    }
+
+    public boolean waitForAssignments(Duration timeout) throws InterruptedException {
+        if (!assignmentLatch.await(timeout.toMillis(), TimeUnit.MILLISECONDS)) {
+            log.warn(metadata.name() + ": Timeout waiting for assignment");
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void run() {
+        threadName = Thread.currentThread().getName();
+        boolean interrupted = false;
+        computation = supplier.get();
+        log.debug(metadata.name() + ": Init");
+        computation.init(context);
+        log.debug(metadata.name() + ": Start");
+        try {
+            processLoop();
+        } catch (InterruptedException e) {
+            // this is expected when the pool is shutdownNow
+            if (log.isTraceEnabled()) {
+                log.debug(metadata.name() + ": Interrupted", e);
+            } else {
+                log.debug(metadata.name() + ": Interrupted");
+            }
+            // the interrupt flag is set after the tailer are closed
+            interrupted = true;
+        } catch (Exception e) {
+            if (Thread.currentThread().isInterrupted()) {
+                // this can happen when pool is shutdownNow throwing ClosedByInterruptException
+                log.info(metadata.name() + ": Interrupted", e);
```

Comment:
first we are not going to ignore an interruption because we are in a process of terminating a thread here (getting out of the run) but it might be useful to put the interrupted flag. 
if interrupted is true we give a chance to destroy and close to be executed before the end of the thread, this is usefull to give faster feedback to kafka which is usefull during test.
I am not for changing anything before having proof it does not work, this code is tested by unit test trying to kill the threadpool under load. though I will add a comment thanks